extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlPatch.java

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.extended.kubectl;
+
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.util.PatchUtils;
+import io.kubernetes.client.util.PatchUtils.PatchCallFunc;
+
+public class KubectlPatch<ApiType extends KubernetesObject>
+    extends Kubectl.ResourceBuilder<ApiType, KubectlPatch<ApiType>>
+    implements Kubectl.Executable<ApiType> {
+  private String patchFormat;
+  private PatchCallFunc patchCallFunc;
+
+  KubectlPatch(Class<ApiType> apiTypeClass) {
+    super(apiTypeClass);
+  }
+
+  public KubectlPatch patchFormat(String patchFormat) {
+    this.patchFormat = patchFormat;
+    return this;
+  }
+
+  public KubectlPatch patchCallFunc(PatchCallFunc patchCallFunc) {
+    this.patchCallFunc = patchCallFunc;
+    return this;
+  }
+
+  @Override
+  public ApiType execute() throws KubectlException {
+    try {
+      return PatchUtils.patch(apiTypeClass, patchCallFunc, patchFormat);
+    } catch (ApiException e) {
+      throw new KubectlException(
```

Comment:
```suggestion
      throw new KubectlException(e)
```

to preserve the stacktraces

Reply:
Done